jctools-benchmarks/src/main/java/org/jctools/jmh/latency/QueueBurstCost.java

```diff
@@ -68,102 +71,145 @@ public void setupQueue()
     {
-        consumer.isRunning = true;
-        consumer.stopped = new CountDownLatch(consumerCount);
+        stopped = new CountDownLatch(consumerCount);
+        final int consumerCount = this.consumerCount;
         for (int i = 0; i < consumerCount; i++)
         {
-            consumerExecutor.execute(consumer);
+            consumers[i].isRunning = true;
+            consumers[i].stopped = stopped;
+        }
+        for (int i = 0; i < consumerCount; i++)
+        {
+            consumerExecutor.execute(consumers[i]);
         }
-
     }
 
     @TearDown(Level.Iteration)
     public void stopConsumers() throws InterruptedException
     {
-        consumer.isRunning = false;
-        consumer.stopped.await();
+        final int consumerCount = this.consumerCount;
+        for (int i = 0; i < consumerCount; i++)
+        {
+            consumers[i].isRunning = false;
+        }
+        stopped.await();
     }
 
     @TearDown(Level.Trial)
-    public void stopExecutor() throws InterruptedException
+    public void stopExecutor()
     {
         consumerExecutor.shutdown();
     }
 
     @Benchmark
     @CompilerControl(CompilerControl.Mode.DONT_INLINE)
-    public void burstCost(Stop stop)
+    public void burstCost(Event event)
     {
         final int burst = burstSize;
-        final Queue<AbstractEvent> q = this.q;
-        final Go go = GO;
-        stop.lazySet(false);
-        sendBurst(q, burst, go, stop);
-
-        waitForIt(stop);
+        final Queue<Event> q = this.q;
+        event.reset();
+        sendBurst(q, event, burst);
+        event.waitFor(burst);
     }
 
-    private void waitForIt(Stop stop)
+    private static void sendBurst(Queue<Event> q, Event event, int burst)
     {
-        while (!stop.get())
+        for (int i = 0; i < burst; i++)
         {
-            ;
+            while (!q.offer(event))
+            {
+                ;
+            }
+            if (DELAY_PRODUCER != 0)
+            {
+                Blackhole.consumeCPU(DELAY_PRODUCER);
+            }
         }
     }
 
-    private void sendBurst(Queue<AbstractEvent> q, int burst, Go go, Stop stop)
+    //do not extends AtomicLong here: JMH will take care of padding fields for us
+    @State(Scope.Thread)
+    public static class Event
     {
-        for (int i = 0; i < burst - 1; i++)
+        private static final long ARRAY_BASE;
+        private static final int ELEMENT_SHIFT;
+
+        static
         {
-            while (!q.offer(go))
+            final int scale = UnsafeAccess.UNSAFE.arrayIndexScale(long[].class);
+            if (8 == scale)
             {
-                ;
+                ELEMENT_SHIFT = 3;
             }
-            if (DELAY_PRODUCER != 0)
+            else
             {
-                Blackhole.consumeCPU(DELAY_PRODUCER);
+                throw new IllegalStateException("Unexpected long[] element size");
             }
+            // Including the buffer pad in the array base offset
```

Comment:
What buffer pad?

Reply:
Ups, copied from:  https://github.com/JCTools/JCTools/blob/172a2f87a2b381cb7bee481df6b72a4abc07a261/jctools-core/src/main/java/org/jctools/queues/ConcurrentSequencedCircularArrayQueue.java#L40

Let me remove it (if the long[] approach will survive)

Reply:
Given that we're going to deal with false sharing through padding, I'm going to resolve this